File: libaries/ts/src/raw.ts
class RawServer {
    private url: string;
    private auth: string;

    private fetch_path: string;

    constructor(url: string, auth: string) {
        this.url = url;
        this.auth = auth;
    }

    private add_params(path: string, params: any): string {
        let result = path;
        let first = true;
        for (let key in params) {
            if (first) {
                result += '?';
                first = false;
            } else {
                result += '&';
            }
            result += key + '=' + params[key];
        }
        return result;
    }

    public async get(path: string): Promise<string> {
        return fetch(this.add_params(this.url + "/get", {
            token: this.auth,
            path: path
        }), {
            method: 'GET'
        }).then((response) => {
            // error code
            if (response.status != 200) {
                throw new Error(response.statusText);
            }
            return response.text();
        });
    }

    public async set(path: string, content: string): Promise<string> {
        return fetch(this.add_params(this.url + "/set", {
            token: this.auth,
            path: path,
            value: content
        }), {
            method: 'PUT'
        }).then((response) => {
            // error code
            console.log(this.url + "/set")
            if (response.status != 200) {
                throw new Error(response.statusText);
            }
            return response.json();
        });
    }

    public async delete(path: string): Promise<string> {
        return fetch(this.add_params(this.url + "/delete", {
            token: this.auth,
            path: path
        }), {
            method: 'DELETE'
        }).then((response) => {
            // error code
            if (response.status != 200) {
                throw new Error(response.statusText);
            }
            return response.json();
        });
    }

    public async data(): Promise<string> {
        return fetch(this.add_params(this.url + "/data", {
            token: this.auth
        }), {
            method: 'GET'
        }).then((response) => {
            // error code
            if (response.status != 200) {
                throw new Error(response.statusText);
            }
            return response.text();
        });
    }

    public async save(): Promise<boolean> {
        return fetch(this.add_params(this.url + "/save_to_file", {
            token: this.auth
        }), {
            method: 'POST'
        }).then((response) => {
            // error code
            if (response.status != 200) {
                throw new Error(response.statusText);
            }
            return true;
        });
    }

    public async load(): Promise<boolean> {
        return fetch(this.add_params(this.url + "/load_from_file", {
            token: this.auth
        }), {
            method: 'POST'
        }).then((response) => {
            // error code
            if (response.status != 200) {
                throw new Error(response.statusText);
            }
            return true;
        });
    }
}

export { RawServer };

File: libraries/cpp/src/cpp/raw.cpp
#include "../include/raw.h"
#include "../include/requests.h"

#include <cpr/cpr.h>



RawServer::RawServer(std::string url, std::string auth)
{
    this->url = url;
    this->auth = auth;
}

RawServer::~RawServer()
{

}

std::string RawServer::get(std::string path)
{
    std::map<std::string, std::string> parameters = {
            { "path", path },
            { "token", auth }
    };

    std::string query = complete_url(url + "/get", parameters);

    return GET(query);
}

std::string RawServer::set(std::string path, std::string value)
{
    std::map<std::string, std::string> parameters = {
            { "path", path },
            { "value", value },
            { "token", auth }
    };

    std::string query = complete_url(url + "/set", parameters);

    return PUT(query);
}

std::string RawServer::del(std::string path)
{
    std::map<std::string, std::string> parameters = {
            { "path", path },
            { "token", auth }
    };

    std::string query = complete_url(url + "/delete", parameters);

    return DEL(query);
}

std::string RawServer::data()
{
    std::map<std::string, std::string> parameters = {
            { "token", auth }
    };

    std::string query = complete_url(url + "/data", parameters);

    return GET(query);
}

std::string RawServer::save()
{
    std::map<std::string, std::string> parameters = {
            { "token", auth }
    };

    std::string query = complete_url(url + "/save_to_file", parameters);

    return POST(query);
}

std::string RawServer::load()
{
    std::map<std::string, std::string> parameters = {
            { "token", auth }
    };

    std::string query = complete_url(url + "/load_from_file", parameters);

    return POST(query);
}

NetworkObject::NetworkObject(RawServer *server, std::string path, std::string *data)
{
    this->server = server;
    this->path = path;
    this->data = data;
}

NetworkObject::~NetworkObject()
{

}

void NetworkObject::set()
{
    server->set(path, *data);
}
